src/main/java/com/gargoylesoftware/htmlunit/javascript/host/file/File.java

```diff
@@ -43,15 +47,38 @@
     private static final String LAST_MODIFIED_DATE_FORMAT = "EEE MMM dd yyyy HH:mm:ss 'GMT'Z (zzzz)";
     private static final String LAST_MODIFIED_DATE_FORMAT_FF = "EEE MMM dd yyyy HH:mm:ss 'GMT'Z";
 
+    private static final String OPTIONS_TYPE_NAME = "type";
+    //default according to https://developer.mozilla.org/en-US/docs/Web/API/File/File
+    private static final String OPTIONS_TYPE_DEFAULT = "";
+    private static final String OPTIONS_LASTMODIFIED = "lastModified";
+
     private java.io.File file_;
 
-    /**
-     * Creates an instance.
-     */
-    @JsxConstructor({CHROME, FF, FF68})
+    private String name;
+
+    private String fileType;
+
+    private long lastModified;
+
+    private long length;
+
     public File() {
     }
 
+    @JsxConstructor({CHROME, FF, FF68})
+    public File(NativeArray bytes, String name, NativeObject options) {
+        this.name = name;
+        this.length = bytes.getLength();
+        
+        this.fileType = options != null ? (String) options.getOrDefault(OPTIONS_TYPE_NAME, OPTIONS_TYPE_DEFAULT) : OPTIONS_TYPE_DEFAULT;
+        String secondsSince1970 = options != null ? (String) options.get(OPTIONS_LASTMODIFIED) : null;
+        if (StringUtils.isNumeric(secondsSince1970)) {
+            this.lastModified = Long.parseLong(secondsSince1970);
+        } else {
+            this.lastModified = System.currentTimeMillis() / 1000;
+        }
+    }
+
     File(final String pathname) {
```

Comment:
In general it might be a better read / easier to understand to have this in a subclass and move the whole java.io.File dependency there.
The option question then would still remain what we'd do with `getFile()` and how an abstract version of this should look like.
Right now I'm not saving the byte array since it's not really used.